### PR TITLE
Support `$rename` field update operator

### LINKED ISSUE #626: Support `$rename` field update operator
https://www.mongodb.com/docs/manual/reference/operator/update/rename/

### DoD

Dot notation should be supported and tested. We should also check if we need to add test cases for dot notation to other compat update tests.

Update documentation about supported operators.

### PR DESCRIPTION
**Pull request by fenogentov**
Closes #626.

### CODE REVIEW

File: integration/update_test.go [hunk 1]
```diff
@@ -460,3 +461,158 @@ func TestUpdateSet(t *testing.T) {
...
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t)
+
+			_, err := collection.InsertMany(ctx, []any{
+				bson.D{{"_id", "1"}, {"name", "alex"}, {"phone", "9012345678"}},
+				bson.D{{"_id", "2"}, {"name", "bob"}},
+			})
+			require.NoError(t, err)
```
Comment (AlekSi) [inline integration/update_test.go:596]: I think that it becomes a problem that, unlike `find` tests, each `update` test uses its own data set. Let's put that PR on pause until we decide if we want to do something about that. Created #674 for tracking

Comment (AlekSi) [reply integration/update_test.go:596]: Let's hook into update_field_compat_test.go instead of using test-specific data set

**Review by AlekSi — CHANGES_REQUESTED** [review]
Please also resolve merge conflicts

File: internal/handlers/common/update.go [hunk 2]
```diff
@@ -113,6 +113,46 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 				}
 			}
 
+		case "$rename":
```
Comment (w84thesun) [inline internal/handlers/common/update.go:116]: Please move that code to the new function `processRenameFieldExpression`.

**Review by w84thesun — CHANGES_REQUESTED** [review]

**Review by rumyantseva — COMMENTED** [review]
Let's make the necessary changes.

**Comment by rumyantseva** [comment]
Supported in #1753.

**Pull request closed without merging**